frontend-react/src/components/header/DAPHeader.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { Helmet } from "react-helmet";
+import { useLocation } from "react-router-dom";
+
+const urlsForDAP = [
+    "/login",
+    "/getting-started/",
+    "/how-it-works/",
+    "/terms-of-service",
+];
+
+export const useDAP = (env: string | undefined) => {
+    const location = useLocation();
+    let currentPathname = location.pathname;
+
+    return (
+        env === "production" &&
+        (currentPathname === "/" ||
+            urlsForDAP.some((url) => currentPathname.startsWith(url)))
+    );
+};
+
+export interface DAPHeaderProps {
+    env: string | undefined;
+}
+
+export const DAPHeader = (params: DAPHeaderProps) => {
+    if (!useDAP(params.env)) return <></>;
+
+    return (
+        <>
+            <Helmet>
+                <script
```

Comment:
I'm curious if any network requests are sent to dap.digital.gov as you click around the SPA.

Reply:
Ugh, @TomNUSDS . With a single comment, you may have shot this ticket dead! 😐🔫
So, it does _not_ send requests as you click around- only with the first visit to the site. Doing the same thing on SimpleReport's site shows network requests going out with each page- probably because it is a _static_ site, with each link click doing a full page load.

So maybe this is just not all that doable on our site, after all. However, there are alternative, way-more-complex approaches for dynamically loaded links suggested in the [DAP Implementation Guide](https://github.com/digital-analytics-program/gov-wide-code/blob/master/documentation/GSA%20DAP%204.1%20-%20Quick%20Guide.pdf) that might work for us. I'll explore these today.

![image](https://user-images.githubusercontent.com/92405130/167180378-cc140323-58ad-4549-8792-17025d2114ff.png)

Reply:
Out of pure curiosity, what happens if you were to move `<DAPHeader>` to the helmet of the component at the url. So instead of being App-level, maybe having it trigger when I arrive at `/getting-started` would help?

Reply:
@kevinhaube na, that bore no fruit. For SNG I even tried putting the DAPHeader at each "page" level- in Overview.tsx, ElrChecklist.tsx, etc. But it's the same behavior- it appropriately writes the <script> to the <head> but it doesn't trigger a network call unless you refresh the page. ☹

Reply:
@kevinhaube @TomNUSDS  Nothing from the guide really panned out for me, but Hack Overflow did present me with a potential way forward. Similar to the idea Kevin was curious about- instead of putting the DAPHeader on the index page, if I call this function at the beginning of the page, it makes the network call every time:

```javascript 
const reloadDAPScript = () => {
    let head= document.getElementsByTagName('head')[0];
    let script= document.createElement('script');
    script.src= 'https://dap.digitalgov.gov/Universal-Federated-Analytics-Min.js';
    script.id = 'testing'
    head.appendChild(script);
}
````
If I mod this so that it removes the existing script before re-adding it, and then find a way to not have to call it on every page you want tracked... maybe it could work? Thoughts? It's still an ugly hack. Might be better to shelve this whole thing...

Reply:
What's the call _for_? I'm not keen on how DAP works. Let's chat about it at eng sync and maybe spitball solutions! One thing I'm thinking is maybe using an effect hook to make the call and reusing that across components as a custom hook.

Reply:
I believe that DAP is actually an older google analytics.

It might be enough to add it to the index page and count how many visitors we get to the site vs individual page views. I think there are fed requirements to use DAP may only care about overall traffic to the site.

We should double check this before doing too much work here.

There's another ticket to track more granular access using Azure App Insights for our own usage.

Reply:
Per our convo in the eng sync today, we're just going to constraint it to the main page for now. This should be good enough for DAP, for now- the App Insights initiative will provide better and more detailed analytics

Reply:
^ which I've now done. also note added explanatory NOTE in DAPHeader.tsx